src/com/boxboat/jenkins/library/git/GitAccount.groovy

```diff
@@ -42,7 +42,18 @@ class GitAccount implements Serializable {
             exit 0
         """
         def checkoutData = Config.pipeline.checkout Config.pipeline.scm
-        return new GitRepo(dir: Utils.toAbsolutePath("."), checkoutData: checkoutData)
+        def repo = new GitRepo(dir: Utils.toAbsolutePath("."), checkoutData: checkoutData)
+
+        if (Config.repo.prUseTargetBranch) {
+            if (Config.pipeline.env?.CHANGE_BRANCH) {
+                repo.setBranch(Config.pipeline.env.CHANGE_BRANCH)
+                repo.setPrBranch(checkoutData?.GIT_BRANCH)
+            }
+        } else if (checkoutData?.GIT_BRANCH) {
+            repo.setBranch(checkoutData?.GIT_BRANCH)
```

Comment:
I think that setting the branch here removes the need to pass `checkoutData` to `GitRepo`, and that `checkoutData` can be removed from `GitRepo` altogether

Reply:
Concur... I'll take it out